src/main/java/seedu/address/model/expense/Amount.java

```diff
@@ -36,7 +36,7 @@ public Amount(String amount) {
      * Returns true if a given string is a valid name.
      */
     public static boolean isValidAmount(String test) {
-        return test.matches(VALIDATION_REGEX);
+        return test.matches(VALIDATION_REGEX) && test != "";
```

Comment:
Do not need to test for != "" because the parser had taken care of blank space. @talkintomato

Reply:
right now i have a test case to test for "" and its failing if I remove this condition I added test!="" an extra check I feel that adding this makes sure that we wont ever have a case where we end up with an amount of "". I dont really see any down side to including this requirement for isValidAmount